### ISSUE TITLE
Improve embedding visualization capabilities

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#46]
The current tester can compute tsne, but it uses the very slow sklearn implementation. UMAP might be a better choice.

**Comment by KevinMusgrave** [comment]
In v0.9.84, you can pass in a ```visualizer``` and ```visualizer_hook``` to BaseTester. The visualizer needs to implement the ```fit``` and ```transform``` function. (In the next version, I'll allow ```fit_transform``` as well.)

For example:
```python
# For plotting the embeddings
def visualizer_hook(umapper, umap_embeddings, labels, split_name, keyname):
    logging.info("UMAP plot for the {} split and label set {}".format(split_name, keyname))
    umap.plot.points(umapper, labels=labels, show_legend=False)
    plt.show()

GlobalEmbeddingSpaceTester(visualizer=umap.UMAP(), visualizer_hook=visualizer_hook)
```

See [the documentation](https://kevinmusgrave.github.io/pytorch-metric-learning/testers/#basetester) and the [Google Colab notebook](https://colab.research.google.com/drive/1fwTC-GRW3X6QiJq6_abJ47On2f3s9e5e) for an example
